core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -1584,6 +1606,59 @@ private ExprValue exprTimeToSec(ExprValue time) {
     return new ExprLongValue(time.timeValue().toSecondOfDay());
   }
 
+  private ExprValue exprTimestampAdd(ExprValue partExpr,
+                                     ExprValue amountExpr,
+                                     ExprValue datetimeExpr) {
+    String part = partExpr.stringValue();
+    int amount = amountExpr.integerValue();
+    LocalDateTime datetime = datetimeExpr.datetimeValue();
+    ChronoUnit temporalUnit;
+
+    switch (part) {
+      case "MICROSECOND":
+        temporalUnit = MICROS;
+        break;
+      case "SECOND":
+        temporalUnit = SECONDS;
+        break;
+      case "MINUTE":
+        temporalUnit = MINUTES;
+        break;
+      case "HOUR":
+        temporalUnit = HOURS;
+        break;
+      case "DAY":
+        temporalUnit = DAYS;
+        break;
+      case "WEEK":
+        temporalUnit = WEEKS;
+        break;
+      case "MONTH":
+        temporalUnit = MONTHS;
+        break;
+      case "QUARTER":
+        temporalUnit = MONTHS;
+        amount *= 3;
+        break;
+      case "YEAR":
+        temporalUnit = YEARS;
+        break;
+      default:
+        throw new SemanticCheckException("Invalid interval argument");
```

Comment:
or NULL?

Reply:
Ah I see we have another issue relating to https://github.com/opensearch-project/sql/issues/1423. I will change it to return `NULL`, but actually it won't change anything for anyone who uses the CLI because an exception will be thrown at the parsing stage.
```
opensearchsql> SELECT TIMESTAMPADD(RANDOM, 1, '2003-01-02 00:00:00');
{'reason': 'Invalid SQL query', 'details': '', 'type': 'NullPointerException'}
```

MySQL also throws an error related to parsing.
```
Error Code: 1064. You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'RANDOM, 1, '2003-01-02 00:00:00')' at line 1
```

An interesting thing is that the `default` statement in the switch/case can still be accessed by tests, so the test `getInvalidTestDataForTimestampAdd` still works properly.

Reply:
A nice confirmation that 100% coverage != everything is tested.
Testing is a tricky science and it requires thorough analysis.

Reply:
Changed and added test in https://github.com/Bit-Quill/opensearch-project-sql/pull/246/commits/77226baed22abe00079b735df28ff26f2cc7547e